dist/tools/whitespacecheck/check.sh

```diff
@@ -27,8 +27,14 @@ if [ -z "${BRANCH}" ]; then
     BRANCH=$(git rev-list HEAD | tail -n 1)
 fi
 
-git diff --check $(git merge-base ${BRANCH} HEAD)
-if [ $? -gt 0 ]
+git -c core.whitespace="tab-in-indent,tabwidth=4" \
+    diff --check $(git merge-base ${BRANCH} HEAD) -- *.[ch]
+
+RESULT=$?
+
+git -c core.whitespace="trailing-space" \
+    diff --check $(git merge-base ${BRANCH} HEAD)
```

Comment:
According to [git-config manpage](http://git-scm.com/book/es/v2/Customizing-Git-Git-Configuration#idp32093488): carriage-returns at line ends are handled as errors in the default case by `--check`. Only if `cr-at-eol` is set, they are allowed.